test/worker_queue_test.dart

```diff
@@ -0,0 +1,122 @@
+import 'package:fluffychat/utils/manager/downloading_worker_queue.dart';
+import 'package:fluffychat/utils/task_queue/task.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:matrix/matrix.dart';
+
+void main() {
+  // Helper function to generate tasks
+  Task generateTask(
+    String taskId,
+    Future Function() runnable, {
+    void Function()? onTaskCompleted,
+  }) {
+    return Task(
+      id: taskId,
+      runnable: runnable,
+      onTaskCompleted: onTaskCompleted,
+    );
+  }
+
+  // Testing group for worker queue
+  group("worker queue test", () {
+    // Test case for adding tasks without cancelation or error
+    test("add 4 tasks without any cancel or error in worker queue", () async {
```

Comment:
need test to verify only one test can run at one time

Reply:
<img width="577" alt="Screenshot 2024-03-18 at 15 45 45" src="https://github.com/linagora/twake-on-matrix/assets/43041967/a792cbd5-943f-4d6a-84e5-13202aa1dbcd">
Added expect() to guaruatee only one task is running at one time